### PR TITLE
env support

### PR DESCRIPTION
**Pull request by HUSKUT**

### CODE REVIEW

File: constants/ApiUrl.ts [hunk 1]
```diff
@@ -1,9 +1,24 @@
-const ApiUrl = 'https://bami-backend.preview.marnixah.com/';
+// @ts-ignore
```
Comment (SophiaH67) [inline constants/ApiUrl.ts]: Use //@ts-expect-error instead

File: constants/ApiUrl.ts [hunk 2]
```diff
@@ -1,9 +1,24 @@
...
+if ({API_URL} == undefined) {
+    ApiEnv = 'https://bami-backend.preview.marnixah.com/'
+} else {
+    ApiEnv = {API_URL}.API_URL
+}
+
+const ApiUrl = ApiEnv.toString()
+
+console.log(ApiUrl)
+// @ts-ignore
```
Comment (SophiaH67) [inline constants/ApiUrl.ts]: use //@ts-expect-error instead

File: lib/signin.ts [hunk 3]
```diff
@@ -9,11 +9,13 @@ export const setToken = async (token: string) => {
 };
 
 export default async function signin(username: string, password: string) {
+  console.log(username, password)
   let bdata = await backendFetch<SignInDTO>('POST', 'api-token-auth/', {
     username: username,
     password: password,
   });
 
+  console.log(bdata)
```
Comment (SophiaH67) [inline lib/signin.ts]: Console.log

File: lib/signin.ts [hunk 4]
```diff
@@ -9,11 +9,13 @@ export const setToken = async (token: string) => {
 };
 
 export default async function signin(username: string, password: string) {
+  console.log(username, password)
```
Comment (SophiaH67) [inline lib/signin.ts]: Remove debug logs?

**Review by SophiaH67 — CHANGES_REQUESTED** [review]

File: constants/ApiUrl.ts [hunk 5]
```diff
@@ -1,9 +1,21 @@
-const ApiUrl = 'https://bami-backend.preview.marnixah.com/';
+//@ts-expect-error
+import { API_URL } from '@env';
+
+let ApiEnv;
+
+if ({ API_URL } == undefined) {
```
Comment (SophiaH67) [inline constants/ApiUrl.ts]: \>creates new object
\>compares it to undefined
image

File: constants/ApiUrl.ts [hunk 6]
```diff
@@ -1,9 +1,21 @@
...
+
+const ApiUrl = ApiEnv.toString();
 
-// If last character is not a slash, raise an error
 if (ApiUrl.slice(-1) !== '/')
   throw new Error(
     'API_URL environment variable must end with a slash. Please update your environment.'
   );
 
+// If last character is not a slash, raise an error
```
Comment (SophiaH67) [inline constants/ApiUrl.ts]: Messed up position of comment?

**Review by SophiaH67 — CHANGES_REQUESTED** [review]

File: constants/ApiUrl.ts (hunk 6 above)
Comment (HUSKUT) [reply constants/ApiUrl.ts]: image

**Review by SophiaH67 — APPROVED** [review]

**Pull request merged**